digitalaidseattle/seattle-humane-app
# Features/0.2.1.1/lint configuration/air bnb

## What type of PR is this? (check all applicable)


## Description

This PR adds lint rules based on the popular AirBnB style guide, along with NextJS lint rules. 

It also includes many changes for auto-fixed lint issues, such as consistent quote usage, consistent indentations, consistent bracket/paren position and spacing, jsx formatting, function declaration styles, and more. 

For rules that could not be auto-fixed, the PR adds blanket exceptions to existing errors (via eslint-disable * comments at the top of the files). Going forward, the next time those files are touched, the dev should remove the exceptions and implement fixes to the issues (which likely includes removing unused code in many cases).

After this change, any lint rule errors will result in build errors. This will enforce the coding practices encouraged by the chosen lint rules. 

## Related Tickets & Documents

- Related Issue #
0.2.1 Linting
- Closes #
0.2.1 Linting

## QA Instructions, Screenshots, Recordings

Run `npm install` to install the new eslint related modules
Run `npm build`
Verify the linting stage is run and completes without error like below:
![image](https://github.com/digitalaidseattle/seattle-humane-app/assets/17752660/d3aacdd4-7250-451d-8d83-2386e5d77c31)

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [x] No, and this is why: The lint rules and auto-fixes do not change product behavior or functionality

## [optional] Are there any post deployment tasks we need to perform?
No, the lint step is already a part of the default NextJS build script.

## [optional] What gif best describes this PR or how it makes you feel?
![sweeping it all under the rug](https://media1.tenor.com/m/a4qkGwHCfxEAAAAC/sweep-sweep-under-rug.gif)
